packages/runtime/container-runtime/src/gcConfig.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IConfigProvider } from "@fluidframework/telemetry-utils";
+import { UsageError } from "@fluidframework/driver-utils";
+import { unreachableCase } from "@fluidframework/common-utils";
+import { getGCVersion, IGCMetadata } from "./summaryFormat";
+import { IGCRuntimeOptions } from "./containerRuntime";
+import { runSessionExpiryKey } from "./garbageCollection";
+
+/** All known test mode configurations */
+export interface IGCTestConfig {
+    /**
+     Sweep V0 is a test mode for using a very short timeout in order to enable testing of real sweep behavior.
+     It's similar to testMode but rather than deleting immediately, it exercises the Sweep timers and codepaths
+    */
+    SweepV0?: {
+        sessionExpiryTimeoutMs: number;
+    };
+}
+
+/** All known test mode names */
+export type GCTestMode = keyof IGCTestConfig;
+
+export const oneDayMs = 1 * 24 * 60 * 60 * 1000;
+
+const defaultCacheExpiryTimeoutMs = 2 * oneDayMs; // 2 days, matches the same variable in odsp-driver
+export const defaultInactiveTimeoutMs = 7 * oneDayMs; // 7 days
+export const defaultSessionExpiryDurationMs = 30 * oneDayMs; // 30 days
+const defaultBufferMs = oneDayMs; // 1 day
+
+export type GcContainerConfig =
+    // Common props regardless of sweepAllowed
+    {
+        prevSummaryGCVersion?: number;
+    }
+    & (
+        // Different sets of props depending on sweepAllowed
+        | {
+            sweepAllowed: false;
+            gcAllowed: boolean;
+        }
+        | {
+            sweepAllowed: true;
+            gcAllowed: true;
+
+            testMode: GCTestMode | undefined;
+            snapshotCacheExpiryMs: number;
+            sessionExpiryTimeoutMs: number;
+            sweepTimeoutBufferMs: number;
+        }
+    );
+
+//* TODO: Add logging of different config sources and outcome (either here or in GC.ts)
+
+export function configForExistingContainer(
+    metadata?: IGCMetadata,
+): GcContainerConfig {
+    // Crash if we see a test mode we don't recognize, since we don't know how it's implemented or should be used
+    const testMode = metadata?.gcTestMode;
+    if (testMode !== undefined && testMode !== "SweepV0") {
+        unreachableCase(testMode, `Cannot open container created under unknown GC Test Mode [${testMode}]`);
```

Comment:
do we have a way to get out of this situation?

Reply:
This is by design. The point is to only allow editing a test document by code that knows about it. In other words, any test mode document is basically considered "corrupted" to clients not aware of how to use it. It's a fail-fast mechanism; a test mode could result in a legit corruption anyway.

With 20-20 hindsight I would make this same change a few versions ago to prevent SweepV0 docs from being opened by older clients.

Reply:
See #11225